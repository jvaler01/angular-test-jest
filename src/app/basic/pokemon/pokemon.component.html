<h2>Pokemon</h2>
<hr>

<div *ngIf="!pokemon; else elseBlock">
    <h3>Loading...</h3>
</div>

<ng-template #elseBlock>
    <h3>{{ pokemon?.name | titlecase }}</h3>
    <img [src]="pokemon?.sprites?.front_default" [alt]="pokemon?.name">
</ng-template>

