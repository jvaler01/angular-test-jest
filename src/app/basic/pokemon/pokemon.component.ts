import { Component, OnInit } from '@angular/core';
import { Pokemon } from '../interfaces';
import { PokemonService } from '../services/pokemon.service';

@Component({
  selector: 'app-pokemon',
  templateUrl: './pokemon.component.html',
  styleUrls: ['./pokemon.component.css']
})
export class PokemonComponent implements OnInit {

  public pokemon?: Pokemon;

  constructor( private pokemonService: PokemonService ) { }

  ngOnInit(): void {
    this.pokemonService.getPokemon(25).subscribe( pokemon =>{
      this.pokemon = pokemon;
      //console.table(this.pokemon);
    })
  }

}
