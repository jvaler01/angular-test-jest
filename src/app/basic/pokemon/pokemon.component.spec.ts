import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';

import { PokemonService } from '../services/pokemon.service';

import { PokemonComponent } from './pokemon.component';

describe('PokemonComponent', () => {
  let component: PokemonComponent;
  let fixture: ComponentFixture<PokemonComponent>;
  let compiled: HTMLElement;
  let service: PokemonService;
  let httpMock: HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ PokemonComponent ],
      imports: [
        HttpClientTestingModule
      ],
      providers: [
        PokemonService
      ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(PokemonComponent);
    component = fixture.componentInstance;
    service = TestBed.inject( PokemonService );
    httpMock = TestBed.inject( HttpTestingController );

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('check snapshoot', () => {
    expect( compiled ).toMatchSnapshot();
  });

  it('check loading', () => {
    const h3 = compiled.querySelector('h3');
    expect( h3?.textContent ).toContain('Loading...');
  });

  it('check get pikachu and load data', () => {

    const dummyPokemon = {
      name: 'pikachu',
      sprites: {
        front_default: 'https://pikachu.com/sprite.png'
      }
    };

    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/25');
    expect( request.request.method ).toBe('GET');
    request.flush( dummyPokemon );

    fixture.detectChanges();
    
    const h3  = compiled.querySelector('h3');
    const img = compiled.querySelector('img');

    expect( h3?.textContent?.toLocaleLowerCase() ).toContain( dummyPokemon.name.toLocaleLowerCase() );
    expect( img?.src ).toBe( dummyPokemon.sprites.front_default );
    expect( img?.alt ).toBe( dummyPokemon.name );


  });
});
