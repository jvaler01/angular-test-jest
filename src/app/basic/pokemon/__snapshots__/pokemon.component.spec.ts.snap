// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonComponent check snapshoot 1`] = `
<div
  id="root1"
>
  <h2>
    Pokemon
  </h2>
  <hr />
  <div>
    <h3>
      Loading...
    </h3>
  </div>
  
  
</div>
`;
